addons/wowl/static/tests/helpers/mock_server.ts

```diff
@@ -101,3 +134,510 @@ export function makeMockServer(
+      const [_model, _viewID, _viewType] = fullKey.split(",");
+      viewId = parseInt(_viewID, 10);
+      return _model === model && _viewType === viewType;
+    });
+    arch = genericViewKey && this.views![genericViewKey];
+  }
+  if (!arch) {
+    throw new Error("No arch found for key " + key);
+  }
+  return {
+    viewId: viewId as number,
+    arch,
+  };
+}
+
+type Operator = string;
+type Modifiers = "invisible" | "readonly" | "required";
+type DomainOp = "|" | "&" | "!";
+type DomainLeaf = DomainOp | [string, Operator, any];
+type Domain = DomainLeaf[];
+
+interface FVG {
+  arch: string;
+  fields: { [fieldName: string]: any };
+  model: string;
+  type: ViewType;
+  toolbar?: boolean;
+  view_id?: number;
+}
+
+function _fieldsViewGet(
+  this: ServerData,
+  arch: string | Element,
+  model: string,
+  fields: { [fieldName: string]: any },
+  context: Context,
+  processedNodes?: Element[]
+): FVG {
+  if (!processedNodes) {
+    processedNodes = [];
+  }
+  function isNodeProcessed(node: Element): boolean {
+    return processedNodes!.findIndex((n) => n.isSameNode(node)) > -1;
+  }
+
+  const modifiersNames: Modifiers[] = ["invisible", "readonly", "required"];
+  const onchanges = this.models![model].onchanges || {};
+  const fieldNodes: {
+    [name: string]: Element;
+  } = {};
+  const groupbyNodes: {
+    [name: string]: Element;
+  } = {};
+
+  let doc: Element;
+  if (typeof arch === "string") {
+    const domParser = new DOMParser();
+    doc = domParser.parseFromString(arch, "text/xml").documentElement;
+  } else {
+    doc = arch;
+  }
+
+  //const inTreeView = (doc.nodeName === 'TREE');
+
+  // mock _postprocess_access_rights
+  const isBaseModel = !context.base_model_name || model === context.base_model_name;
+  const views = ["kanban", "tree", "form", "gantt", "activity"];
+  if (isBaseModel && views.indexOf(doc.tagName) !== -1) {
+    for (const action of ["create", "delete", "edit", "write"]) {
+      if (!doc.getAttribute(action) && action in context && !context[action]) {
+        doc.setAttribute(action, "false");
+      }
+    }
+  }
+
+  traverseElementTree(doc, (node) => {
+    if (node.nodeType === Node.TEXT_NODE) {
+      return false;
+    }
+    const modifiers: {
+      [name: string]: Domain | any;
+    } = {};
+
+    const isField = node.nodeName === "FIELD";
+    const isGroupby = node.nodeName === "GROUPBY";
+
+    if (isField) {
+      const fieldName = node.getAttribute("name")!;
+      fieldNodes[fieldName!] = node;
+
+      // 'transfer_field_to_modifiers' simulation
+      const field = fields[fieldName];
+
+      if (!field) {
+        throw new Error("Field " + fieldName + " does not exist");
+      }
+      const defaultValues: any = {};
+      const stateExceptions: any = {}; // what is this ?
+      modifiersNames.forEach((attr) => {
+        stateExceptions[attr] = [];
+        defaultValues[attr] = !!field[attr];
+      });
+      // LPE: what is this ?
```

Comment:
attribute `state="open,pending"` stands for `invisible if the state isn't in [open, pending]`